resin/models/application.py

```diff
@@ -217,27 +222,27 @@ def restart(self, name):
             endpoint=self.settings.get('api_endpoint'), login=True
         )
 
-    def get_api_key(self, name):
+    def get_config(self, app_id):
         """
-        Get the API key for a specific application. This function only works if you log in using credentials or Auth Token.
+        Download application config.json.
 
         Args:
             name (str): application name.
 
         Returns:
-            str: API key.
+            dict: application config.json content.
 
         Raises:
             ApplicationNotFound: if application couldn't be found.
 
         Examples:
-            >>> resin.models.application.get_api_key('RPI1')
-            u'XbKn5GhK4YieOLpX4KjQTqjqo1moRWmP'
-
+            >>> resin.models.application.get_config('106640')
+            {u'applicationName': u'RPI3', u'username': u'nghiant2710', u'apiKey': u'kIaqS6ZLOoxkFzpzqSYhWtr2lj6m8KZi', u'vpnPort': 443, u'listenPort': 48484, u'pubnubSubscribeKey': u'sub-c-bbc12eba-ce4a-11e3-9782-02ee2ddab7fe', u'vpnEndpoint': u'vpn.resin.io', u'userId': 189, u'files': {u'network/network.config': u'[service_home_ethernet]\nType = ethernet\nNameservers = 8.8.8.8,8.8.4.4', u'network/settings': u'[global]\nOfflineMode=false\nTimeUpdates=manual\n\n[WiFi]\nEnable=true\nTethering=false\n\n[Wired]\nEnable=true\nTethering=false\n\n[Bluetooth]\nEnable=true\nTethering=false'}, u'pubnubPublishKey': u'pub-c-6cbce8db-bfd1-4fdf-a8c8-53671ae2b226', u'apiEndpoint': u'https://api.resin.io', u'connectivity': u'connman', u'deviceType': u'raspberrypi3', u'mixpanelToken': u'99eec53325d4f45dd0633abd719e3ff1', u'deltaEndpoint': u'https://delta.resin.io', u'appUpdatePollInterval': 60000, u'applicationId': 106640, u'registryEndpoint': u'registry.resin.io'}
         """
 
-        app = self.get(name)
+        # Application not found will be raised if can't get app by id.
+        self.get_by_id(app_id)
         return self.base_request.request(
-            'application/{0}/generate-api-key'.format(app['id']), 'POST',
-            endpoint=self.settings.get('api_endpoint'), login=True
+            '/download-config?appId={id}'.format(id=app_id), 'GET',
```

Comment:
@pimterry, my bad. Since copied the API endpoint from this commit https://github.com/resin-io/resin-sdk/commit/35363e97ae4a4445bbc6f7e7fa8e5716260e26e2 so didn't notice that. Will update it now